Description: Don't start smbd when guest account doesn't exist
 When the guest account doesn't exist, smbd should not start, because it will
 crash when accessing a share as guest user.
 .
 This patch correctly returns the status of make_new_server_info_guest.
 .
 This is a backport of commit 51567de6d7d91f727a2135cd2d5434f85db6d45a by
 Volker Lendecke <vl@samba.org> from upstream samba git master.
Author: Ivo De Decker <ivo.dedecker@ugent.be>
Origin: upstream
Bug: https://bugzilla.samba.org/show_bug.cgi?id=9004
Bug-Debian: http://bugs.debian.org/653382
Forwarded: not-needed
Last-Update: 2012-06-24

--- samba-3.6.5.orig/source3/auth/auth_util.c
+++ samba-3.6.5/source3/auth/auth_util.c
@@ -760,7 +760,7 @@ static NTSTATUS make_new_server_info_gue
 	status = NT_STATUS_OK;
 done:
 	TALLOC_FREE(tmp_ctx);
-	return NT_STATUS_OK;
+	return status;
 }
 
 /***************************************************************************
